maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenTestsPlugin.java

```diff
@@ -0,0 +1,118 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import java.io.File;
+import java.util.List;
+
+import org.apache.maven.model.Resource;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+
+import aQute.bnd.osgi.Builder;
+import aQute.bnd.osgi.Constants;
+
+@Mojo(name = "bnd-process-tests", defaultPhase = LifecyclePhase.PROCESS_TEST_CLASSES, requiresDependencyResolution = ResolutionScope.COMPILE, threadSafe = true)
+public class BndMavenTestsPlugin extends AbstractBndMavenPlugin {
+
+	/**
+	 * If true, make the tests artifact of fragment of the main artifact.
+	 */
+	@Parameter(defaultValue = "false")
+	private boolean									fragment;
+
+	/**
+	 * Possible values are {@link TestCases#junit3 junit3},
+	 * {@link TestCases#junit4 junit4}, {@link TestCases#junit5 junit5} and
+	 * {@link TestCases#all all}
+	 */
+	@Parameter(defaultValue = "junit4")
+	private TestCases								testCases;
+
+	@Parameter(defaultValue = "${project.build.testSourceDirectory}", readonly = true)
+	private File									sourceDir;
+
+	@Parameter(defaultValue = "${project.build.testResources}", readonly = true)
+	private List<org.apache.maven.model.Resource>	resources;
+
+	@Parameter(defaultValue = "${project.build.outputDirectory}")
+	private File									mainClassesDir;
+
+	@Parameter(defaultValue = "${project.build.testOutputDirectory}")
+	private File									classesDir;
+
+	@Parameter(defaultValue = "${project.build.testOutputDirectory}")
+	private File									outputDir;
+
+	@Parameter(defaultValue = "${project.build.testOutputDirectory}/META-INF/MANIFEST.MF")
+	private File									manifestPath;
+
+	@Parameter(property = "bnd-tests.skip", defaultValue = "false")
+	private boolean									skip;
+
+	@Override
+	public File getSourceDir() {
+		return sourceDir;
+	}
+
+	@Override
+	public List<Resource> getResources() {
+		return resources;
+	}
+
+	@Override
+	public File getClassesDir() {
+		return classesDir;
+	}
+
+	@Override
+	public File getOutputDir() {
+		return outputDir;
+	}
+
+	@Override
+	public File getManifestPath() {
+		return manifestPath;
+	}
+
+	@Override
+	public boolean isSkip() {
+		return skip;
+	}
+
+	@Override
+	protected void processBuilder(Builder builder) {
+		String bsn = builder.getProperty(Constants.BUNDLE_SYMBOLICNAME);
+		builder.setProperty(Constants.BUNDLE_SYMBOLICNAME, bsn + "-tests");
+		builder.setProperty(Constants.TESTCASES, testCases.filter());
```

Comment:
Wrap `testCases.filter()` with `${sort;${uniq;` here.

Reply:
sure why not...